src/pydot/dot_parser.py

```diff
@@ -113,57 +113,21 @@ def parseImpl(
         )
 
 
-def push_top_graph_stmt(
-    s: str, loc: int, toks: ParseResults
-) -> list[pydot.core.Dot] | pydot.core.Dot:
-    attrs = {}
+def push_top_graph_stmt(toks: ParseResults) -> list[pydot.core.Dot]:
     top_graphs = []
-    g: pydot.core.Dot = None  # type: ignore
-
-    for element in toks:
-        if (
-            isinstance(element, (ParseResults, tuple, list))
-            and len(element) == 1
-            and isinstance(element[0], str)
-        ):
-            element = element[0]
-
-        if element == "strict":
-            attrs["strict"] = True
-
-        elif element in ["graph", "digraph"]:
-            attrs = {}
-
-            g = pydot.core.Dot(graph_type=element, **attrs)
-            attrs["type"] = element
 
-            top_graphs.append(g)
+    for result in toks.graphs:
+        assert isinstance(result, ParseResults)
+        gtype = result.gtype
+        strict = "strict" in result
+        id_ = result.id or ""
 
-        elif isinstance(element, str):
-            g.set_name(element)
-
-        elif isinstance(element, pydot.core.Subgraph):
-            g.obj_dict["attributes"].update(element.obj_dict["attributes"])
-            g.obj_dict["edges"].update(element.obj_dict["edges"])
-            g.obj_dict["nodes"].update(element.obj_dict["nodes"])
-            g.obj_dict["subgraphs"].update(element.obj_dict["subgraphs"])
-
-            g.set_parent_graph(g)
-
-        elif isinstance(element, P_AttrList):
-            attrs.update(element.attrs)
-
-        elif isinstance(element, (ParseResults, list)):
-            add_elements(g, element)
-
-        else:
-            raise ValueError(f"Unknown element statement: {element}")
-
-    for g in top_graphs:
+        g = pydot.core.Dot(id_, graph_type=gtype, strict=strict)
+        g.set_parent_graph(g)
+        if isinstance(result.contents, ParseResults):
```

Comment:
(Same reason this check is here, though in this case `result.contents` actually could come back as `''` if it's parsing an empty `{ }`.)